### PR TITLE
Add shared library

### PR DESCRIPTION
**Pull request by mattiamencagli**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,10 +3,22 @@ cmake_minimum_required(VERSION 3.10)
 set(LJMD_VERSION 1.0 CACHE STRING "Floating point number for the current version")
 project(ljmd VERSION ${LJMD_VERSION} LANGUAGES C)
 
-add_executable(ljmd.x main.c src/utilities.c src/input_output.c src/engine.c )
+add_executable(ljmd.x
```
Comment (lfabris-mhpc) [inline CMakeLists.txt]: I think that short lines like this one are readable as-is and don't need breaking

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,10 +3,22 @@ cmake_minimum_required(VERSION 3.10)
...
+	main.c
+)
 
-target_include_directories(ljmd.x PRIVATE "include")
+add_library(ljmd_rt SHARED
```
Comment (lfabris-mhpc) [inline CMakeLists.txt:7]: these one instead deserved to have its parameters split one per line

File: CMakeLists.txt [hunk 3]
```diff
@@ -3,10 +3,22 @@ cmake_minimum_required(VERSION 3.10)
...
+    src/engine.c
+    src/input_output.c
+    src/utilities.c
+)
+
+target_include_directories(ljmd.x PRIVATE include)
```
Comment (lfabris-mhpc) [inline CMakeLists.txt]: in these multiple target_* lines, I think we should separate those regarding the lib and those regarding the exe
so, for example: first do all commands for the lib, then all commands for the exe

**Comment by mattiamencagli** [comment]
I agree with your comments, after dinner I will change these details

**Comment by mattiamencagli** [comment]
Ok, After some tries now this pull request to the master has the changes that you implement Lorenzo.

**Review by lfabris-mhpc — APPROVED** [review]

**Review by otroconi — APPROVED** [review]

**Pull request merged**